From 4a03ea7d1f8a5c2e387f98990236a2479a811c50 Mon Sep 17 00:00:00 2001
From: Ruslan Akbashev <ra@turingpi.com>
Date: Thu, 13 Apr 2023 20:55:16 +0400
Subject: [PATCH] Makefile: use xxd to generate C headers

Build process of usbboot requires running a built binary (bin2c) on a
host machine, which is not possible when cross-compiling. Meanwhile,
output of xxd is identical to one produced by b2c, so use that.
---
 Makefile | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index fa324ab..720ee18 100755
--- a/Makefile
+++ b/Makefile
@@ -1,3 +1,5 @@
+XXD = xxd -i -c 16
+
 rpiboot: main.c msd/bootcode.h msd/start.h msd/bootcode4.h msd/start4.h
 	$(CC) -Wall -Wextra -g -o $@ $< -lusb-1.0
 
@@ -8,10 +10,10 @@ librpiboot.a: main.o
 	$(AR) rcs $@ $<
 
 %.h: %.bin ./bin2c
-	./bin2c $< $@
+	$(XXD) $< $@
 
 %.h: %.elf ./bin2c
-	./bin2c $< $@
+	$(XXD) $< $@
 
 bin2c: bin2c.c
 	$(CC) -Wall -Wextra -g -o $@ $<
-- 
2.39.2

